Fix force checkin button

njohner opened the pull request:
Buttons get defined on the class level, so conditionally changing the button label is not a good idea. Instead we now have 2 buttons that are displayed conditionally.
I also added the French translation of the button label.

**Before, once the label was changed to "checking anyway", it could remain in that state even if the document is not locked anymore**
![screen shot 2018-03-19 at 14 18 23](https://user-images.githubusercontent.com/7374243/37597712-7517acd6-2b80-11e8-9bd7-7a9b9bfe8227.png)

**Now  we get the correct labels**
![screen shot 2018-03-19 at 13 56 20](https://user-images.githubusercontent.com/7374243/37597544-06b45dd4-2b80-11e8-8fa7-db03dcf69d73.png)
![screen shot 2018-03-19 at 13 57 16](https://user-images.githubusercontent.com/7374243/37597545-06d6afe2-2b80-11e8-843a-55021476a8c8.png)
**And the french translation**
<img width="961" alt="screen shot 2018-03-19 at 14 02 09" src="https://user-images.githubusercontent.com/7374243/37597554-0a6b32cc-2b80-11e8-90e9-7d8d4d0da7cd.png">


resolves https://extranet.4teamwork.ch/support/edk/tracker-support-gever-edk/102

jone commented:
Could you add a screenshot? This could help giving some context 😉

njohner commented:
I had already made them, but forgot to add them to the PR

njohner commented:
Nooo it failed. I guess I'll need to fix the tests :-)

phgross commented on opengever/document/checkout/checkin.py:
```diff
@@ -115,23 +114,27 @@ def __init__(self, context, request):
 
-    @button.buttonAndHandler(_(u'button_checkin', default=u'Checkin'))
-    def checkin_button_handler(self, action):
+    def document_is_locked(self):
```
I would rename it to `is_document_locked`

phgross commented on opengever/document/checkout/checkin.py:
```diff
@@ -115,23 +114,27 @@ def __init__(self, context, request):
+
+    @button.buttonAndHandler(_(u'button_checkin', default=u'Checkin'),
+                             condition=lambda form: not form.document_is_locked())
+    def checkin_button_handler(self, action, condition=lambda form: not form.document_is_locked()):
```
Is it necessary to repeat the condition in the method itself? According to the [docs](https://docs.plone.org/develop/addons/schema-driven-forms/customising-form-behaviour/actions.html#conditional-actions) it's not.

njohner replied:
Oh no, sorry. That is a "typo", or rather a copy paste fail.

njohner commented on opengever/document/checkout/checkin.py:
```diff
@@ -115,23 +114,27 @@ def __init__(self, context, request):
+                             condition=lambda form: not form.document_is_locked())
+    def checkin_button_handler(self, action, condition=lambda form: not form.document_is_locked()):
+        # Errors are handled by the checkin
+        data = self.extractData()[0]
```
done

njohner commented:
* @phgross I hadn't done that translation because I was having some issues translating the last sentence. I added one now, but I feel it's not perfect. As I wrote in the gever channel, I think we need a real translation of `checkout` in French.
* I added a test to check that the proper buttons are displayed
* Corrected another translation (the warning said the document was "locked" instead of saying that it is checked-out)
* In the French warning "Avec check-out" means "with checkout", which is not was is meant here ("checked out"). I am unable to come up with a translation as "checkout" cannot be used as a verb in French. If we use something else, we would need to change it everywhere in gever. (I think "retirer" would be a good option if we want to go down that path)

**French view of force checkin**
<img width="935" alt="screen shot 2018-03-19 at 15 34 09" src="https://user-images.githubusercontent.com/7374243/37604436-443b660a-2b91-11e8-923b-0de8d86eeb2e.png">

deiferni commented on opengever/document/tests/test_checkout.py:
```diff
@@ -335,6 +336,31 @@ def get_manager(self, document):
+
+        # Lock document
+        IRefreshableLockable(self.document).lock()
+        transaction.commit()
```
you should never have to commit in  an `IntegrationTestCase`

njohner commented:
@deiferni Ok everything done
